sqlglot/tokens.py

```diff
@@ -327,6 +327,7 @@ class TokenType(AutoName):
     PREWHERE = auto()
     PRIMARY_KEY = auto()
     PROCEDURE = auto()
+    PROJECTION = auto()
```

Comment:
We prefer not adding new tokens for cases like this one
```suggestion
```

Reply:
Is there a rule for cases like this? I'll try to remember to check the dialect for parser dicts now, but is there something else to keep in mind?

Reply:
My rule of thumb is to not add tokens so I can simply do `self._match`, because the `TokenType` enum will explode and that will create more complexity in recognizing their purpose / adding them to the right mappings in the parser.

Tokens are generally needed when needing to add a token to an existing mapping, eg. type tokens, `NO_PAREN_FUNCTION` tokens, etc.

Reply:
Will need that in mind, thanks!